javascript/javascript1/week4/voice.js

```diff
@@ -0,0 +1,260 @@
+      if (
+        splittedArray[i].toLowerCase() === "what" &&
+        splittedArray[i + 1].toLowerCase() === "is"
+      ) {
+        switch (splittedArray[i + 3]) {
+          case "+":
+            response =
+              Number(splittedArray[i + 2]) + Number(splittedArray[i + 4]);
+            break;
+          case "plus":
+            response =
+              Number(splittedArray[i + 2]) + Number(splittedArray[i + 4]);
+            break;
+          case "-":
+            response =
+              Number(splittedArray[i + 2]) - Number(splittedArray[i + 4]);
+            break;
+          case "minus":
+            response =
+              Number(splittedArray[i + 2]) - Number(splittedArray[i + 4]);
+            break;
+          case "*":
+            response =
+              Number(splittedArray[i + 2]) * Number(splittedArray[i + 4]);
+            break;
+          case "times":
+            response =
+              Number(splittedArray[i + 2]) * Number(splittedArray[i + 4]);
+            break;
+          case "/":
+            response =
+              Number(splittedArray[i + 2]) / Number(splittedArray[i + 4]);
+            break;
+          case "slash":
+            response =
+              Number(splittedArray[i + 2]) / Number(splittedArray[i + 4]);
+            break;
+        }
+      }
+    }
+  }
+  //case6 set a timer
+  else if (
+    command.toLowerCase().includes("set") ||
+    command.toLowerCase().includes("timer")
+  ) {
+    for (let i = 0; i < splittedArray.length; i++) {
+      if (splittedArray[i].toLowerCase().includes("for")) {
+        if (
+          splittedArray[i].toLowerCase().includes("1") ||
+          splittedArray[i].toLowerCase().includes("one")
+        ) {
+          response = `Timer set for ${splittedArray[i + 1]} minut`;
+        } else {
+          response = `Timer set for ${splittedArray[i + 1]} minutes`;
+        }
+        const milliseconds = Number(splittedArray[i + 1]) * 60 * 1000;
+        const print = () => (response = "Timer done");
+        setTimeout(print, milliseconds);
+      }
+    }
+  }
+
+  //final method 8 ball
+  else {
+    response =
+      eightBallAnswers[Math.floor(Math.random() * eightBallAnswers.length)];
+  }
+  //final return
+  return response.toString();
+};
+
+function isgetReplyAvailable() {
+  return typeof getReply !== "undefined" && typeof getReply === "function";
+}
+
+if (isgetReplyAvailable()) {
+  let command;
+  let timeoutId;
+  let setIntervalTimer;
+
+  const button = document.querySelector("button");
+  button.addEventListener("click", () => {
+    button.innerHTML = "Talk now 🙂";
+    setIntervalTimer = setInterval(() => {
+      const randomNumber = Math.floor(Math.random() * 6) + 2;
+      if (randomNumber % 2 === 0) {
+        button.innerHTML = "Talk now 😮";
+      } else {
+        button.innerHTML = "Talk now 🙂";
+      }
+    }, 100);
+    clearTimeout(timeoutId);
+
+    command = "";
+    timeoutId = setTimeout(() => {
```

Comment:
Adding a `try...catch` block for error handling in asynchronous functions is definitely a good practice. 

It helps ensure that any errors occurring during asynchronous operations are properly caught and handled, preventing them from crashing the application or causing unexpected behavior. This can significantly improve the robustness and reliability of the code.

Reply:
i did it, but try ... catch is still new ground for me